provider/osio/osio.go

```diff
@@ -147,15 +148,23 @@ func (p *Provider) loadRules(clusterResp *clusterResponse) *types.Configuration
 
 	defaultBackendExist := false
 	for ind, cluster := range clusterResp.Clusters {
-		if p.defaultBackendURL != "" && p.defaultBackendURL == cluster.APIURL {
+		if p.defaultBackendURL != "" && p.defaultBackendURL == getDefaultURL(cluster) {
 			defaultBackendExist = true
 		}
 		configInd := fmt.Sprintf("%d", ind+1)
-		config.Frontends["frontend"+configInd] = createFrontend(cluster.APIURL, "backend"+configInd)
-		config.Backends["backend"+configInd] = createBackend(cluster.APIURL)
+		if cluster.APIURL != "" {
+			config.Frontends["api"+configInd] = createFrontend(cluster.APIURL, "api"+configInd)
+			config.Backends["api"+configInd] = createBackend(cluster.APIURL)
+		}
+		if cluster.MetricsURL != "" {
+			config.Frontends["metrics"+configInd] = createFrontend(cluster.MetricsURL, "metrics"+configInd)
+			config.Backends["metrics"+configInd] = createBackend(cluster.MetricsURL)
+		}
```

Comment:
Previously, frontend/backend added to support api_url.  Now we have api_url and metrics_url to which oso-proxy forward the calls.  So now separate config added for api and metrics along with default config.